onnxruntime/core/session/inference_session.h

```diff
@@ -263,7 +263,7 @@ class InferenceSession {
   common::Status Run(const RunOptions& run_options, const std::vector<std::string>& feed_names,
                      const std::vector<OrtValue>& feeds, const std::vector<std::string>& output_names,
                      std::vector<OrtValue>* p_fetches,
-                     const std::vector<OrtDevice>* p_fetches_device_info = nullptr) ORT_MUST_USE_RESULT;
+                     const std::vector<OrtDevice>* p_fetches_device_info = nullptr, int64_t run_id = DEFAULT_RUN_ID) ORT_MUST_USE_RESULT;
```

Comment:
>int64_t run_id = DEFAULT_RUN_ID [](start = 84, length = 31)

Let's add a comment here explaining run_id should be kept as default for non-partial run. 

run_id will be used to lookup previous incomplete runs for Partition Run.